src/utils/storage.ts

```diff
@@ -0,0 +1,53 @@
+import * as vscode from "vscode";
+
+import { addToCache, getFromCache } from "./addonCache";
+import { createComment } from "./comment";
+import getCommentLocation, { stringToRange } from "./getThreadLocation";
+import { loadFileDecorator } from "./loadComments";
+import { splitUri } from "./splitUri";
+import { AssayComment } from "../config/comment";
+
+export async function fetchCommentsFromCache(
+  controller: vscode.CommentController
+) {
+  const workspace = vscode.workspace.workspaceFolders;
+  if (!workspace) {
+    return;
+  }
+
+  const uri = workspace[0].uri;
+  const { rootFolder, fullPath, guid } = await splitUri(uri);
+
+  if (!fullPath.startsWith(rootFolder)) {
+    return;
+  }
+
+  const comments = await getFromCache(guid, ["comments"]);
+
+  for (const version in comments) {
+    for (const filepath in comments[version]) {
+      for (const lineNumber in comments[version][filepath]) {
+        const { uri, body, contextValue } =
+          comments[version][filepath][lineNumber];
+        const r = stringToRange(lineNumber);
+        const thread = controller.createCommentThread(uri, r, []);
+        createComment(contextValue, new vscode.MarkdownString(body), thread);
+      }
+    }
+  }
```

Comment:
This is probably too much business logic related to the comment data model for a general purpose "storage" file. This feels like it should be refactored into a `getAllComments()` method or even something that implements the [iterable protocol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#the_iterable_protocol) (e.g. `CommentManager.entries()` a la [`Map.prototype.entries()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/entries)).